SYCL/Matrix/joint_matrix_query_use_default.cpp

```diff
@@ -0,0 +1,166 @@
+//==------ joint_matrix_query_use_default.cpp  - DPC++ joint_matrix---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: matrix
+
+// RUN: %clangxx -fsycl %s -o %t.out -DSYCL_EXT_ONEAPI_MATRIX_VERSION=2
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
+public:
+  T *mat;
+
+public:
+  T *get_data() { return mat; }
+  void set_data(T *data) { mat = data; }
+  big_matrix(T *data) : mat(data) {}
+};
+
+template <typename T1, typename T2, size_t NUM_ROWS_A, size_t NUM_COLS_A,
+          size_t NUM_ROWS_B, size_t NUM_COLS_B, size_t NUM_ROWS_C,
+          size_t NUM_COLS_C>
+void matrix_multiply(big_matrix<T1, NUM_ROWS_C, NUM_COLS_C> &C,
+                     big_matrix<T2, NUM_ROWS_A, NUM_COLS_A> &A,
+                     big_matrix<T2, NUM_ROWS_B, NUM_COLS_B> &B) {
+  size_t M = NUM_ROWS_C;
+  size_t N = NUM_COLS_C;
+  size_t K = NUM_COLS_A;
+  assert(NUM_ROWS_C == NUM_ROWS_A && NUM_COLS_A == NUM_ROWS_B * 4);
+
+  using myparams2 = tpu_params<tpu::amx, int8_t, int8_t, int>;
+  constexpr int TM = myparams2::M;
+  constexpr int TN = myparams2::N;
+  constexpr int TK = myparams2::K;
+
+  std::cout << "AMX query sizes are: M " << TM << " N " << TN << " K " << TK
```

Comment:
check the query size as well

Reply:
@yanfeng3721 if there is "failed" in line48 and another "passed" in line166, can the test infrastructure detect it is failed?

Reply:
> @yanfeng3721 if there is "failed" in line48 and another "passed" in line166, can the test infrastructure detect it is failed?

Hi @yubingex007-a11y, I think it can be checked by FileCheck. Please refer to usage doc https://llvm.org/docs/CommandGuide/FileCheck.html. 
Also, there is an example test as reference: https://github.com/intel/llvm-test-suite/blob/intel/SYCL/DiscardEvents/discard_events_usm.cpp